src/recipe.js

```diff
@@ -10,14 +10,17 @@ class Recipe {
     this.instructions = recipe.instructions
   }
 
+
   calculateIngredientsCost(ingredientData) {
-   let costData = ingredientData.filter(ingredient => this.ingredients.some(recipeIngredient => recipeIngredient.id === ingredient.id));
```

Comment:
Let's check the consistency of how this function performs. I'm curious to see what problems it's causing in the DOM.